addons/sale_project/models/project.py

```diff
@@ -118,6 +126,11 @@ def _compute_display_sales_stat_buttons(self):
         for project in self:
             project.display_sales_stat_buttons = project.allow_billable and project.partner_id
 
+    @api.onchange('allow_billable')
+    def _onchange_allow_billable(self):
+        if not self.allow_billable and self.billing_type == 'manually':
+            self.billing_type = 'not_billable'
```

Comment:
I think it could be a compute method instead. To be sure, the billing_type changes if the allow_billable is altered in the code and not only in the form view of `project.project` model. 🙂